### ISSUE TITLE
Echo in Windows 10 and Android, no echo on iOS and iPadOS.

### ISSUE THREAD

**Issue by Ogglas** [microsoft/cognitive-services-speech-sdk-js#207]
The following code is triggered once, `console.log(text)` is printed once anyway.

Using microsoft-cognitiveservices-speech-sdk 1.12.1 I hear the text with an eco on the following devices:

Windows 10 Chrome and Firefox
Android  - Internet, Brave and Firefox

No echo on these devices:
iOS with Firefox, Safari and Chrome
iPadOS with Firefox, Safari and Chrome

What could be the cause of this?

```
import * as sdk from 'microsoft-cognitiveservices-speech-sdk';
import { SpeechSynthesizer } from 'microsoft-cognitiveservices-speech-sdk';
import { getToken } from '../api/microsoftCognitiveServicesSpeechController';

export const textToSpeech = async (text: string) => {
    console.log(text);
    let token = await (await getToken()).data;
    let soundContext: AudioContext | undefined = undefined;
    try {
        var AudioContext = window.AudioContext || (window as any).webkitAudioContext || false;
        if (AudioContext) {
            soundContext = new AudioContext();
        }
        else {
            alert("AudioContext not supported");
        }
    }
    catch (e) {
        console.log("no sound context found, no audio output. " + e);
    }

    let speechConfig = sdk.SpeechConfig.fromAuthorizationToken(token, "northeurope");

    speechConfig.speechSynthesisLanguage = "sv-SE";
    speechConfig.speechSynthesisVoiceName = "sv-SE-HedvigRUS";

    let synthesizer: SpeechSynthesizer | undefined = new sdk.SpeechSynthesizer(speechConfig);

    synthesizer.speakTextAsync(text,
        (result) => {
            if (result.audioData && soundContext) {
                let source = soundContext.createBufferSource();
                soundContext.decodeAudioData(result.audioData,
                    (newBuffer) => {
                        source.buffer = newBuffer;
                        source.connect((soundContext as AudioContext).destination);
                        source.start(0);
                    });
            }

            synthesizer?.close();
            synthesizer = undefined;
        },
        (err) => {
            console.log(err);
            synthesizer?.close();
            synthesizer = undefined;
        }
    );
};
```

**Comment by yulin-li** [comment]
The sdk will do playback itself when default audioConfig is used. And your codes try to play again. Just delete the playback codes.

```javascript
          if (result.audioData && soundContext) {
                let source = soundContext.createBufferSource();
                soundContext.decodeAudioData(result.audioData,
                    (newBuffer) => {
                        source.buffer = newBuffer;
                        source.connect((soundContext as AudioContext).destination);
                        source.start(0);
                    });
            }
```

P.S. it seems our sample is outdate, we will update it to correct.

**Comment by Ogglas** [comment]
Thank you @yulin-li! When will you update the sample code?

Tried deleting the code below but then I could not hear anything.

```
          if (result.audioData && soundContext) {
                let source = soundContext.createBufferSource();
                soundContext.decodeAudioData(result.audioData,
                    (newBuffer) => {
                        source.buffer = newBuffer;
                        source.connect((soundContext as AudioContext).destination);
                        source.start(0);
                    });
            }
```

After that I tried removing this but with the same result.

    let soundContext: AudioContext | undefined = undefined;
    try {
        var AudioContext = window.AudioContext || (window as any).webkitAudioContext || false;
        if (AudioContext) {
            soundContext = new AudioContext();
        }
        else {
            alert("AudioContext not supported");
        }
    }
    catch (e) {
        console.log("no sound context found, no audio output. " + e);
    }

**Comment by yulin-li** [comment]
hmm... Could you try https://yulin-test.azurewebsites.net/ to see if it works for you?

**Comment by Ogglas** [comment]
@yulin-li That site leads me to a login screen only.

**Comment by yulin-li** [comment]
The attached is our up-to-date sample, I confirmed it works for me, pls have a try.
[text-to-speech.zip](https://github.com/microsoft/cognitive-services-speech-sdk-js/files/4913720/text-to-speech.zip)

**Comment by Ogglas** [comment]
No sound at all. `result.reason` was `sdk.ResultReason.SynthesizingAudioCompleted` though. 

This was my new code, tested on Windows 10 with Chrome.

```
export const textToSpeech = async (text: string) => {

    let token = await (await getToken()).data;

    let speechConfig = sdk.SpeechConfig.fromAuthorizationToken(token, "northeurope");

    speechConfig.speechSynthesisLanguage = "sv-SE";
    speechConfig.speechSynthesisVoiceName = "sv-SE-HedvigRUS";

    let synthesizer: SpeechSynthesizer | undefined = new sdk.SpeechSynthesizer(speechConfig);

    synthesizer.speakTextAsync(
        text,
        function (result) {
            if (result.reason === sdk.ResultReason.SynthesizingAudioCompleted) {
                console.log(result);
            } else if (result.reason === sdk.ResultReason.Canceled) {
                console.log(result.errorDetails);
            }
            synthesizer?.close();
            synthesizer = undefined;
        },
        function (err) {
            console.log(err);
            synthesizer?.close();
            synthesizer = undefined;
        });
};
```

![image](https://user-images.githubusercontent.com/4015237/87396291-e8466700-c5b2-11ea-9ee5-0d689da5d482.png)

However using the code you sent worked, even when modified for a bit. The only real difference I can see is "Format PCM is not supported for playback." warning in my code. Do you have any idea what this could be?

```
<!DOCTYPE html>
<html lang="en">
<head>
    <title>Microsoft Cognitive Services Speech SDK JavaScript Quickstart</title>
    <meta charset="utf-8" />
</head>
<body style="font-family:'Helvetica Neue',Helvetica,Arial,sans-serif; font-size:13px;">
    <div id="warning">
        <h1 style="font-weight:500;">Speech Recognition Speech SDK not found (microsoft.cognitiveservices.speech.sdk.bundle.js missing).</h1>
    </div>

    <div id="content" style="display:none">
        <table width="100%">
            <tr>
                <td></td>
                <td><h1 style="font-weight:500;">Microsoft Cognitive Services Speech SDK JavaScript Quickstart</h1></td>
            </tr>
            <tr>
                <td></td>
                <td><button id="startSpeakTextAsyncButton">Start Text to Speech</button></td>
            </tr>
            <tr>
                <td align="right" valign="top">Result</td>
                <td><textarea id="resultDiv" style="display: inline-block;width:500px;height:100px"></textarea></td>
            </tr>
        </table>
    </div>

    <!-- Speech SDK reference sdk. -->
    <script src="microsoft.cognitiveservices.speech.sdk.bundle.js"></script>

    <!-- Speech SDK USAGE -->
    <script>
        // status fields and start button in UI
        var resultDiv;
        var startSpeakTextAsyncButton;

        var SpeechSDK;
        var synthesizer;

        document.addEventListener("DOMContentLoaded", function () {
            startSpeakTextAsyncButton = document.getElementById("startSpeakTextAsyncButton");
            resultDiv = document.getElementById("resultDiv");

            startSpeakTextAsyncButton.addEventListener("click", function () {
                startSpeakTextAsyncButton.disabled = true;

                var speechConfig;
                speechConfig = SpeechSDK.SpeechConfig.fromSubscription(<SUBSCRIPTIONKEY>, "northeurope");

                speechConfig.speechSynthesisLanguage = "sv-SE";
                speechConfig.speechSynthesisVoiceName = "sv-SE-HedvigRUS";

                synthesizer = new SpeechSDK.SpeechSynthesizer(speechConfig);
                var text = "Testar"
                synthesizer.speakTextAsync(
                    text,
                    function (result) {
                        startSpeakTextAsyncButton.disabled = false;
                        if (result.reason === SpeechSDK.ResultReason.SynthesizingAudioCompleted) {
                            resultDiv.innerHTML += "synthesis finished for [" + text + "].\n";
                        } else if (result.reason === SpeechSDK.ResultReason.Canceled) {
                            resultDiv.innerHTML += "synthesis failed. Error detail: " + result.errorDetails + "\n";
                        }
                        window.console.log(result);
                        synthesizer.close();
                        synthesizer = undefined;
                    },
                    function (err) {
                        startSpeakTextAsyncButton.disabled = false;
                        resultDiv.innerHTML += "Error: ";
                        resultDiv.innerHTML += err;
                        resultDiv.innerHTML += "\n";
                        window.console.log(err);

                        synthesizer.close();
                        synthesizer = undefined;
                    });
            });

            if (!!window.SpeechSDK) {
                SpeechSDK = window.SpeechSDK;
                startSpeakTextAsyncButton.disabled = false;

                document.getElementById('content').style.display = 'block';
                document.getElementById('warning').style.display = 'none';
            }
        });

    </script>
</body>
</html>
```


![image](https://user-images.githubusercontent.com/4015237/87399247-63aa1780-c5b7-11ea-97f6-49ab6f68c1cd.png)

**Comment by yulin-li** [comment]
`Format PCM is not supported for playback` is a waring in previous release. make sure you are using the latest (1.12.1) SDK. You can donwload it from [here](https://aka.ms/csspeech/jsbrowserpackage).

**Comment by Ogglas** [comment]
![image](https://user-images.githubusercontent.com/4015237/87402587-11b7c080-c5bc-11ea-8c7d-f0b1e2b48aba.png)

That was it. I was testing the previous version and reverted `package.json` but forgot to run `npm install ` again after that. Thank you very much for your help!

**Comment by yulin-li** [comment]
You are welcome and thanks for reporting this. We just fixed the playback issue in 1.12.1 😄
